src/test/java/com/Mattheo992/medicalclinic/service/PatientServiceTest.java

```diff
@@ -0,0 +1,127 @@
+package com.Mattheo992.medicalclinic.service;
+
+import com.Mattheo992.medicalclinic.model.Patient;
+import com.Mattheo992.medicalclinic.model.User;
+import com.Mattheo992.medicalclinic.model.dtos.PatientDto;
+import com.Mattheo992.medicalclinic.model.mappers.PatientDtoMapper;
+import com.Mattheo992.medicalclinic.model.mappers.UserMapper;
+import com.Mattheo992.medicalclinic.repository.PatientRepository;
+import com.Mattheo992.medicalclinic.repository.UserRepository;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mapstruct.Mapper;
+import org.mapstruct.factory.Mappers;
+import org.mockito.Mockito;
+import org.springframework.boot.autoconfigure.data.web.SpringDataWebProperties;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.when;
+
+public class PatientServiceTest {
+    PatientService patientService;
+    PatientRepository patientRepository;
+    UserRepository userRepository;
+    PatientDtoMapper patientDtoMapper;
+    UserMapper userMapper;
+
+    @BeforeEach
+    void setup() {
+        this.patientRepository = Mockito.mock(PatientRepository.class);
+        this.userRepository = Mockito.mock(UserRepository.class);
+        this.patientDtoMapper = Mappers.getMapper(PatientDtoMapper.class);
+        this.userMapper = Mappers.getMapper(UserMapper.class);
+        this.patientService = new PatientService(patientRepository, userRepository,patientDtoMapper,userMapper);
+    }
+
+    @Test
+    void getPatients_PatientsExist_PatientsReturned() {
+        //given
+        List<Patient> patients = new ArrayList<>();
+        patients.add(createPatient("aa@wp.pl", 1L));
+        patients.add(createPatient("bb@wp.pl", 2L));
+        Pageable pageable = PageRequest.of(0, 10, Sort.by("id").ascending());
+        //when
+        when(patientRepository.findAll()).thenReturn(patients);
```

Comment:
to nie powinno byc w sekcji when -> zauwaz ze sekcja given sluzy do utworzenia danych do  przeprowadzeia testow wlicza sie w to rowniez powiedzenie ktore mocki maja co zwrocic stad taka rzecz powinna byc w sekcji given -> jest to przygotowanie do przeprowadzenia testu